test/WrappedMToken.t.sol

```diff
@@ -237,6 +347,122 @@ contract WrappedMTokenTests is Test {
         assertEq(_wrappedMToken.totalEarningSupply(), 0);
     }
 
+    function testFuzz_unwrap(
+        bool earningEnabled_,
+        bool accountEarning_,
+        uint240 balance_,
+        uint240 unwrapAmount_,
+        uint128 accountIndex_,
+        uint128 currentIndex_
+    ) external {
+        accountEarning_ = earningEnabled_ && accountEarning_;
+
+        if (earningEnabled_) {
+            _registrar.setListContains(_EARNERS_LIST, address(_wrappedMToken), true);
+            _wrappedMToken.enableEarning();
+        }
+
+        accountIndex_ = uint128(bound(accountIndex_, _EXP_SCALED_ONE, 10 * _EXP_SCALED_ONE));
+        balance_ = uint240(bound(balance_, 0, _getMaxAmount(accountIndex_)));
+
+        if (accountEarning_) {
+            _wrappedMToken.setAccountOf(_alice, balance_, accountIndex_);
+            _wrappedMToken.setTotalEarningSupply(balance_);
+
+            _wrappedMToken.setPrincipalOfTotalEarningSupply(
+                IndexingMath.getPrincipalAmountRoundedDown(balance_, accountIndex_)
+            );
+        } else {
+            _wrappedMToken.setAccountOf(_alice, balance_);
+            _wrappedMToken.setTotalNonEarningSupply(balance_);
+        }
+
+        currentIndex_ = uint128(bound(currentIndex_, accountIndex_, 10 * _EXP_SCALED_ONE));
+        unwrapAmount_ = uint240(bound(unwrapAmount_, 0, 2 * balance_));
+
+        _mToken.setCurrentIndex(_currentIndex = currentIndex_);
+
+        uint240 accruedYield_ = _wrappedMToken.accruedYieldOf(_alice);
+
+        _mToken.setBalanceOf(address(_wrappedMToken), balance_ + accruedYield_);
+
+        if (unwrapAmount_ == 0) {
+            vm.expectRevert(abi.encodeWithSelector(IERC20Extended.InsufficientAmount.selector, (0)));
+        } else if (unwrapAmount_ > balance_ + accruedYield_) {
+            vm.expectRevert(
+                abi.encodeWithSelector(
+                    IWrappedMToken.InsufficientBalance.selector,
+                    _alice,
+                    balance_ + accruedYield_,
+                    unwrapAmount_
+                )
+            );
+        }
+
+        vm.startPrank(_alice);
+        _wrappedMToken.unwrap(_alice, unwrapAmount_);
+
+        if ((unwrapAmount_ == 0) || (unwrapAmount_ > balance_)) return;
```

Comment:
Should also include the accrued yield.
```suggestion
        if ((unwrapAmount_ == 0) || (unwrapAmount_ > balance_ + accruedYield_)) return;
```

Reply:
this line is redundant by checks in 389-400

Reply:
On 389, it only checks that the event has been properly emitted. You still need to return early on line 405.